packages/wrangler/src/pages/deployment-tails.tsx

```diff
@@ -137,34 +138,54 @@ export async function Handler({
 		PAGES_CONFIG_CACHE_FILENAME
 	);
 	const accountId = await requireAuth(pagesConfig);
+	let deploymentId = deployment;
 
 	projectName = projectName || pagesConfig.project_name;
 
-	if (!deploymentId && !projectName && isInteractive) {
+	// When not specifying a deployment, we need a project name to use the latest
+	// Or when specifying by deployment URL
+	if ((!deployment || isUrl(deployment)) && !projectName && isInteractive) {
```

Comment:
I wonder...is there any use case in which not specifying the project name shouldn't trigger this prompt?
The `tail`ing request is sent to `/account/<account_id>/pages/projects/<project_name>/deployments`, which I makes `projectName` mandatory, unless we can somehow deduct it based on the params being passed to the `tail` command?

Reply:
noticing that somewhere before this code we set `projectName = projectName || pagesConfig.project_name`, which makes me think we assume? that if no project name is passed as an argument, 
```
const pagesConfig = getConfigCache<PagesConfigCache>(
		PAGES_CONFIG_CACHE_FILENAME
	);
	```
will always have the answers for us. But from what I'm seeing, that doesn't always seem to be the case 👇 

<img width="340" alt="Screenshot 2022-11-11 at 13 01 41" src="https://user-images.githubusercontent.com/4638332/201336556-22028c30-5f15-4cd7-82b1-b360d7761f37.png">

Reply:
I've cleaned this up. It's better to just not cache the project name at all. Only account ID